tests/test_connection.py

```diff
@@ -220,50 +224,82 @@
-#            'fake', 'fake', 'fake', 'fake')
-#        self.connection._connected = True
-#
-#    def test_run_command_exception_eof(self):
-#        """Testing EOFException in run_command."""
-#        self.connection._telnet = mock.Mock()
-#        self.connection._telnet.write = mock.Mock()
-#        self.connection._telnet.write.side_effect = EOFError('except')
-#        self.connection.run_command('test')
-#        self.assertFalse(self.connection._connected)
-#
-#    def test_run_command_exception_connection_refused(self):
-#        """Testing ConnectionRefusedError in run_command."""
-#        self.connection._telnet = mock.Mock()
-#        self.connection._telnet.write = mock.Mock()
-#        self.connection._telnet.write.side_effect = ConnectionRefusedError(
-#            'except')
-#        self.connection.run_command('test')
-#        self.assertFalse(self.connection._connected)
-#
-#    def test_run_command_exception_gaierror(self):
-#        """Testing socket.gaierror in run_command."""
-#        self.connection._telnet = mock.Mock()
-#        self.connection._telnet.write = mock.Mock()
-#        self.connection._telnet.write.side_effect = socket.gaierror('except')
-#        self.connection.run_command('test')
-#        self.assertFalse(self.connection._connected)
-#
-#    def test_run_command_exception_oserror(self):
-#        """Testing OSError in run_command."""
-#        self.connection._telnet = mock.Mock()
-#        self.connection._telnet.write = mock.Mock()
-#        self.connection._telnet.write.side_effect = OSError('except')
-#        self.connection.run_command('test')
-#        self.assertFalse(self.connection._connected)
\ No newline at end of file
+
+
+class TestTelnetConnection(TestCase):
+    """Testing TelnetConnection."""
+
+    def setUp(self):
+        """Set up test env."""
+        self.connection = TelnetConnection(
+                'fake', 'fake', 'fake', 'fake')
+        # self.connection._connected = True
+
+    def test_determine_linelength_inf(self):
+        """ Test input for infinite breakline length."""
+        # An input without newlines results in infinite linebreak
+        # The input string is shorter than the limit
+        for i in (5, 10, 50):
+            input_bytes = (" " * i).encode('ascii')
+            self.connection._determine_linelength(input_bytes)
+            self.assertEqual(self.connection._linebreak, float('inf'))
+
+    def test_determine_linelength(self):
+        for i in (5, 10, 50):
+            input_bytes = (" " * i + "\n" + " " * 5).encode('ascii')
+            self.connection._determine_linelength(input_bytes)
+            self.assertEqual(self.connection._linebreak, i)
+
+            # And now with some more lines
+            input_bytes = ((" " * i + "\n") * 3 + " " * 5).encode('ascii')
+            self.connection._determine_linelength(input_bytes)
+            self.assertEqual(self.connection._linebreak, i)
+
+    @pytest.mark.skip(
```

Comment:
I think you should comment out all the ones below here, they are not ready to be readded